lib/chef/knife/bootstrap.rb

```diff
@@ -650,7 +652,23 @@ def connect!
           end
 
           opts.merge! force_ssh_password_opts(password)
-          do_connect(opts)
+          connect!(opts)
+        else
+          raise
```

Comment:
Best if we could have redefine [exceptions](https://github.com/chef/chef/blob/master/lib/chef/exceptions.rb) and can raise and catch them.

Reply:
that bit is actually fine, the code is just stating that it handles certain Train:Error's but not others.